docs/sql-migration-guide-upgrade.md

```diff
@@ -138,6 +138,8 @@ license: |
     need to specify a value with units like "30s" now, to avoid being interpreted as milliseconds; otherwise, 
     the extremely short interval that results will likely cause applications to fail.
 
+  - When turning a Dataset to another Dataset, Spark will up cast the fields in the original Dataset to the type of corresponding fields in the target DataSet. In version 2.4 and earlier, this up cast is not very strict, e.g. `Seq("str").toDS.as[Int]` fails, but `Seq("str").toDS.as[Boolean]` works and throw NPE during execution. In Spark 3.0, the up cast is stricter and turning String into something else is not allowed, i.e. `Seq("str").toDS.as[Boolean]` will fail during analysis.
```

Comment:
@cloud-fan is it on purpose not to mention setting `spark.sql.legacy.looseUpcast=true` to preserve old behavior?

Reply:
Yeah we should

Reply:
are you interested in creating a PR (with a new JIRA) to fix the doc? 2.4 is EOL but I think it's good to fix the docs anyway.

Reply:
Sure. I will create one.

Reply:
https://github.com/apache/spark/pull/35519 has been created.